src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.ClientCertificateProvider.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+
+using CURLcode = Interop.Http.CURLcode;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler : HttpMessageHandler
+    {
+        internal sealed class ClientCertificateProvider : IDisposable
+        {
+            internal readonly GCHandle _gcHandle;
+            internal readonly Interop.Ssl.ClientCertCallback _callback;
```

Comment:
No, I need to store the values that are returned by the callback to the native library, and will have to dispose them later on.